stl/inc/algorithm

```diff
@@ -504,6 +504,49 @@ _NODISCARD _FwdIt adjacent_find(_ExPo&& _Exec, const _FwdIt _First, const _FwdIt
 
 #ifdef __cpp_lib_concepts
 namespace ranges {
+    // VARIABLE ranges::adjacent_find
+    class _Adjacent_find_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <forward_iterator _It, sentinel_for<_It> _Se, class _Pj = identity,
+            indirect_binary_predicate<projected<_It, _Pj>, projected<_It, _Pj>> _Pr = ranges::equal_to>
+        _NODISCARD constexpr _It operator()(_It _First, const _Se _Last, _Pr _Pred = {}, _Pj _Proj = {}) const {
+            // find first satisfying _Pred with successor
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst = _Get_unwrapped(_First);
+            const auto _ULast  = _Get_unwrapped(_Last);
+
+            if (_UFirst == _ULast) {
+                _Seek_wrapped(_First, _UFirst);

```

Comment:
This seek is unnecessary (because no iterators have been incremented yet). Although this consistently seeks before returning, I was unable to find any precedent for this. Should we drop the `_Seek_wrapped` and comment that `_First` can be returned without seeking?

Reply:
Yes, I think we should drop the `_Seek_wrapped`; the (unwritten) contract for `_Get_unwrapped` is that `_Get_unwrapped(lvalue)` is non-modifying, whereas `_Get_unwrapped(rvalue)` may modify `rvalue`.